lib/charms/haproxy/v0/haproxy_route.py

```diff
@@ -0,0 +1,543 @@
+    rise: int = Field(
+        description="How many successful health checks before server is considered up.", default=2
+    )
+    fall: int = Field(
+        description="How many failed health checks before server is considered down.", default=3
+    )
+    paths: list[str] = Field(description="The health check path.", default=[])
+
+
+# tarpit is not yet implemented
+class RateLimitPolicy(Enum):
+    """Enum of possible http_route types.
+
+    Attrs:
+        DENY: deny a client's HTTP request to return a 403 Forbidden error.
+        REJECT: closes the connection immediately without sending a response.
+        NOPROXY: disconnects immediately without notifying the client that the connection has been closed.
+    """
+
+    DENY = "deny"
+    REJECT = "reject"
+    SILENT = "silent-drop"
+
+
+class RateLimit(BaseModel):
+    connections_per_minute: int = Field(description="How many connections are allowed per minute.")
+    policy: RateLimitPolicy = Field(
+        description="Configure the rate limit policy.", default=RateLimitPolicy.DENY
+    )
+
+
+class LoadBalancingAlgorithm(Enum):
+    """Enum of possible http_route types.
+
+    Attrs:
+        LEASTCONN: The server with the lowest number of connections receives the connection.
+        SRCIP: Load balance using the hash of The source IP address.
+        ROUNDROBIN: Each server is used in turns, according to their weights.
+        COOKIE: Load balance using hash req.cookie(clientid).
+    """
+
+    LEASTCONN = "leastconn"
+    SRCIP = "source"
+    ROUNDROBIN = "roundrobin"
+    COOKIE = "cookie"
+
+
+class LoadBalancingConfiguration(BaseModel):
+    algorithm: LoadBalancingAlgorithm = Field(
+        description="Configure the load balancing algorithm for the service."
+    )
+    cookie: Optional[str] = Field(
+        description="Only used when algorithm is COOKIE. Define the cookie to load balance on.",
+        default=None,
+    )
+
+
+class BandwidthLimit(BaseModel):
+    upload: int = Field(description="Upload limit (bytes per seconds).")
+    download: int = Field(description="Download limit (bytes per seconds).")
+
+
+# retry-on is not yet implemented
+class Retry(BaseModel):
+    count: int = Field(description="How many times should a request retry.")
+    interval: int = Field(description="Interval (in seconds) between retries.")
+    redispatch: bool = Field(
+        description="Whether to redispatch failed requests to another server.", default=False
+    )
+
+
+class TimeoutConfiguration(BaseModel):
+    server: int = Field(description="Timeout for requests from haproxy to backend servers.")
+    client: int = Field(description="Timeout for client requests to haproxy.")
+    queue: int = Field(
+        description="Timeout for requests waiting in the queue after server-maxconn is reached."
+    )
+
+
+class HaproxyRewriteMethod(Enum):
+    """Enum of possible http_route types.
```

Comment:
minor: http_route?

Reply:
I updated the docstring for this, thanks!